_posts/2025-07-25-model-registry-integration.md

```diff
@@ -0,0 +1,445 @@
+
+
+Troubleshooting `make kind-cluster-agnostic`: 
+If you encounter "No rule to make target" or other errors, ensure your `kubeflow/pipelines` repository clone is up-to-date and contains this target in `backend/Makefile`.
+
+Verify KFP Health:
+- After `make kind-cluster-agnostic` finishes, open a new terminal.
+- Run this command repeatedly until ALL (or almost all) pods in the `kubeflow` namespace are `Running` and `READY` `1/1`.
+
+```
+kubectl get pods -n kubeflow
+```
+
+- ***This is crucial. Do not proceed until KFP is healthy.***
+
+
+Clone Kubeflow Model Registry Repository:
+```
+cd ~
+git clone https://github.com/kubeflow/model-registry.git
+```
+
+Deploy Kubeflow Model Registry (API Server): This deploys the Model Registry API server, which stores and serves model metadata.
+```
+cd ~/model-registry/manifests/kustomize/overlays/db # Navigate to the correct overlay for embedded DB
+kubectl apply -k . -n kubeflow
+```
+
+
+- Troubleshooting Model Registry Pods: Wait for `model-registry-deployment pod` to become `Running` and `READY` (`kubectl get pods -n kubeflow | grep model-registry`).
+
+**Install and Deploy Kubeflow Model Registry UI:**
+/ The Model Registry UI is a separate frontend application that provides a browsable interface to your registered models. There are two primary ways to deploy it: a simpler method for basic setups, and a more comprehensive method for multi-user Kubeflow environments.
+
+- **Option A:** Basic UI Deployment (Recommended for simplicity with standalone KFP): This method deploys the UI without requiring Istio or other multi-user components.
+
+Installation (Cloning the `kubeflow/manifests` repository, pinned to a compatible release): First, clone the [Kubeflow manifests repository](https://github.com/kubeflow/manifests), (if you haven't already). We recommend pinning to a specific release (e.g., v1.7.0) compatible with KFP 2.5.0.
+```
+cd ~
+git clone https://github.com/kubeflow/manifests.git kubeflow-manifests-repo # Clone to a distinct name
+cd kubeflow-manifests-repo
+git fetch --tags
+git checkout tags/v1.7.0 # Pin to Kubeflow manifests v1.7.0 (compatible with KFP 2.5.0)
+```
+
+- Deployment (Applying manifests): Navigate to the Model Registry UI `base` overlay:
+```
+cd ~/kubeflow-manifests-repo/applications/model-registry/upstream/options/ui/base # Using 'base' overlay for non-Istio
+```
+
+Apply the UI manifests:
+```
+kubectl apply -k . -n kubeflow
+```
+
+Wait for the UI pod to be ready:
+```
+kubectl get pods -n kubeflow -l app=model-registry-ui
+```
+
+- **Option B:** Multi-User Kubeflow Deployment (More comprehensive, but complex): This option deploys a full multi-user Kubeflow environment, which includes Istio, Dex, and other components necessary for authentication and traffic management. This path is significantly more time-consuming and complex to set up, but it enables the Model Registry UI to integrate with the Central Dashboard in a production-like, authenticated manner.
```

Comment:
I personally would just go with this option in this section to keep things simple.